contracts/pricers/WstethPricer.sol

```diff
@@ -0,0 +1,80 @@
+import {WSTETHInterface} from "../interfaces/WSTETHInterface.sol";
+import {SafeMath} from "../packages/oz/SafeMath.sol";
+
+/**
+ * @notice A Pricer contract for a wstETH token
+ */
+contract WstethPricer is OpynPricerInterface {
+    using SafeMath for uint256;
+
+    /// @notice opyn oracle address
+    OracleInterface public oracle;
+
+    /// @notice wstETH token
+    WSTETHInterface public wstETH;
+
+    /// @notice underlying asset (WETH)
+    address public underlying;
+
+    /**
+     * @param _wstETH wstETH
+     * @param _underlying underlying asset for wstETH
+     * @param _oracle Opyn Oracle contract address
+     */
+    constructor(
+        address _wstETH,
+        address _underlying,
+        address _oracle
+    ) public {
+        require(_wstETH != address(0), "WstethPricer: wstETH address can not be 0");
+        require(_underlying != address(0), "WstethPricer: underlying address can not be 0");
+        require(_oracle != address(0), "WstethPricer: oracle address can not be 0");
+
+        wstETH = WSTETHInterface(_wstETH);
+        oracle = OracleInterface(_oracle);
+        underlying = _underlying;
+    }
+
+    /**
+     * @notice get the live price for the asset
+     * @dev overrides the getPrice function in OpynPricerInterface
+     * @return price of 1e18 stETH in USD, scaled by 1e8
+     */
+    function getPrice() external view override returns (uint256) {
+        uint256 underlyingPrice = oracle.getPrice(address(underlying));
+        require(underlyingPrice > 0, "WstethPricer: underlying price is 0");
+        return _underlyingPriceToWstethPrice(underlyingPrice);
+    }
+
+    /**
+     * @notice set the expiry price in the oracle
+     * @dev requires that the underlying price has been set before setting a wstETH price
+     * @param _expiryTimestamp expiry to set a price for
+     */
+    function setExpiryPriceInOracle(uint256 _expiryTimestamp) external {
+        (uint256 underlyingPriceExpiry, ) = oracle.getExpiryPrice(underlying, _expiryTimestamp);
+        require(underlyingPriceExpiry > 0, "WstethPricer: underlying price not set yet");
+        uint256 wstEthPrice = _underlyingPriceToWstethPrice(underlyingPriceExpiry);
+        oracle.setExpiryPrice(address(wstETH), _expiryTimestamp, wstEthPrice);
+    }
+
+    /**
+     * @dev convert underlying price to wstETH price with the wstETH to stETH exchange rate (1stETH = 1 ETH)
+     * @param _underlyingPrice price of 1 underlying token (ie 1e18 WETH) in USD, scaled by 1e8
+     * @return price of 1e8 wstETH in USD, scaled by 1e8
+     */
+    function _underlyingPriceToWstethPrice(uint256 _underlyingPrice) private view returns (uint256) {
+        uint256 stEthPerWsteth = wstETH.stEthPerToken();
+
+        return stEthPerWsteth.mul(_underlyingPrice).div(1e18);
+    }
+
+    function getHistoricalPrice(uint80 _roundId) external view override returns (uint256, uint256) {
+        revert("WstethPricer: Deprecated");
+    }
+}
```

Comment:
Can you optimize the `require` & `revert` strings like we did for the [`StakedaoEcrvPricer`](https://github.com/opynfinance/perp-vault-templates/blob/stakedao-master/contracts/utils/StakedaoEcrvPricer.sol#L12-L21)?

Here is the [diff](https://github.com/opynfinance/perp-vault-templates/pull/38/files#diff-fc97ea9e7fd83c47df786ce52cf7ecf6f0bb6219b11353942d38c67f01f57539R9).

Reply:
Done